include/net/cloud.h

```diff
@@ -43,6 +44,14 @@ enum cloud_event_type {
 	CLOUD_EVT_COUNT
 };
 
+enum cloud_disconnected_err_type {
+	CLOUD_DISCONNECT_ERR_USER_REQUEST,
```

Comment:
I'd drop the `_ERR` piece in those enums, I wouldn't identify disconnection as an error condition.

Reply:
i added the `_ERR` just because this value is stored in the `data.err` field of `struct cloud_event`